src/core/utils/src/functions/4C_utils_function_manager.cpp

```diff
@@ -80,33 +92,35 @@ void Core::Utils::add_valid_builtin_functions(Core::Utils::FunctionManager& func
       all_of({
           entry<int>("VARIABLE"),
           entry<std::string>("NAME"),
-          entry<std::string>("TYPE"),
-          entry<int>("NUMPOINTS", {.required = false}),
-          tag("BYNUM", {.default_value = false}),
-          entry<std::vector<double>>("TIMERANGE", {.required = false, .size = 2}),
-          entry<std::vector<double>>(
-              "TIMES", {.required = false, .size = from_parameter<int>("NUMPOINTS")}),
-          entry<std::vector<double>>(
-              "VALUES", {.required = false, .size = from_parameter<int>("NUMPOINTS")}),
-          entry<std::vector<std::string>>(
-              "DESCRIPTION", {.required = false,
-                                 .size =
-                                     [](const IO::InputParameterContainer& container)
-                                 {
-                                   try
-                                   {
-                                     return container.get<int>("NUMPOINTS") - 1;
-                                   }
-                                   catch (const Core::Exception& e)
-                                   {
-                                     // When NUMPOINTS is not set, then we still allow for a
-                                     // single DESCRIPTION entry
-                                     return 1;
-                                   }
-                                 }}),
-          tag("PERIODIC", {.default_value = false}),
-          entry<double>("T1", {.required = false}),
-          entry<double>("T2", {.required = false}),
+          one_of({
```

Comment:
This example nicely demonstrates how much more expressive the new InputSpec can be. All the implicit dependencies between the parameters are now made explicit. Instead of having a bunch of optional parameters, we now nicely group the parameters that should be entered together.

Best thing: with the work of @gilrrei this will soon be available through your editor if you use yaml files.